pyro/contrib/tracking/extended_kalman_filter.py

```diff
@@ -0,0 +1,237 @@
+
+    @property
+    def dynamic_model(self):
+        '''
+        Dynamic model access.
+        '''
+        return self._dynamic_model
+
+    @property
+    def dimension(self):
+        '''
+        Native state dimension access.
+        '''
+        return self._dynamic_model.dimension
+
+    @property
+    def mean(self):
+        '''
+        Native state estimate mean access.
+        '''
+        return self._mean
+
+    @property
+    def cov(self):
+        '''
+        Native state estimate covariance access.
+        '''
+        return self._cov
+
+    @property
+    def dimension_pv(self):
+        '''
+        PV state dimension access.
+        '''
+        return self._dynamic_model.dimension_pv
+
+    @property
+    def mean_pv(self):
+        '''
+        Compute and return cached PV state estimate mean.
+        '''
+        if self._mean_pv_cache is None:
+            self._mean_pv_cache = \
+                self._dynamic_model.mean2pv(self._mean)
+
+        return self._mean_pv_cache
+
+    @property
+    def cov_pv(self):
+        '''
+        Compute and return cached PV state estimate covariance.
+        '''
+        if self._cov_pv_cache is None:
+            self._cov_pv_cache = \
+                self._dynamic_model.cov2pv(self._cov)
+
+        return self._cov_pv_cache
+
+    @property
+    def time(self):
+        '''
+        State time access.
+        '''
+        return self._time
+
+    def init(self, mean, cov, time):
+        '''
+        Re-initialize target state.
+
+        :param mean: target state mean.
+        :param cov: target state covariance.
+        :param time: state time. None => keep existing time.
+        '''
+        self._mean = mean.clone()
+        self._cov = cov.clone()
+        if time is not None:
+            self._time = time
+
+        self._clear_cached()
+
+    def predict(self, dt=None, destination_time=None):
+        '''
+        Use dynamic model to predict (aka propagate aka integrate) state
+        estimate in-place.
+
+        :param dt: time to integrate over. The state time will be automatically
+                   incremented this amount unless you provide ``destination_time``.
+                   Using ``destination_time`` may be preferable for prevention of
+                   roundoff error accumulation.
+        :param destination_time: time to increment to.
+        '''
+        if dt is not None and destination_time is not None:
+            assert ((destination_time - self._time + dt) < 1e-3).all()
+        elif dt is None:
+            assert destination_time is not None
+            dt = destination_time - self._time
+        elif destination_time is None:
+            assert dt is not None
+            destination_time = self._time + dt
+
+        self._mean = self._dynamic_model(self._mean, dt)
+
+        F = self._dynamic_model.jacobian(dt)
+        Q = self._dynamic_model.process_noise_cov(dt)
+        self._cov = F.mm(self._cov).mm(F.t()) + Q
```

Comment:
`torch.t()` is intentionally broken to keep compatibility with Numpy's `.T` operator: `x.t() == x.transpose(0, 1)`, whereas to support broadcasting, it should be `x.transpose(-1, -2)`.

Could you replace `.t()` with `.transpose(-1, -2)` throughout, so we're closer to being able to support broadcasting?